src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/telemetry/telemetry.component.ts

```diff
@@ -0,0 +1,203 @@
+  ) {}
+
+  ngOnInit() {
+    this.loading = true;
+    const observables = [
+      this.mgrModuleService.getOptions('telemetry'),
+      this.mgrModuleService.getConfig('telemetry')
+    ];
+    observableForkJoin(observables).subscribe(
+      (resp: object) => {
+        this.moduleEnabled = resp[1]['enabled'];
+        this.options = _.pick(resp[0], this.requiredFields);
+        const configs = _.pick(resp[1], this.requiredFields);
+        this.createConfigForm();
+        this.configForm.setValue(configs);
+        this.loading = false;
+      },
+      (_error) => {
+        this.error = true;
+      }
+    );
+  }
+
+  private createConfigForm() {
+    const controlsConfig = {};
+    _.forEach(Object.values(this.options), (option) => {
+      controlsConfig[option.name] = [option.default_value, this.getValidators(option)];
+    });
+    this.configForm = this.formBuilder.group(controlsConfig);
+  }
+
+  private createPreviewForm() {
+    const controls = {
+      report: JSON.stringify(this.report, null, 2),
+      reportId: this.reportId,
+      licenseAgrmt: [this.licenseAgrmt, Validators.requiredTrue]
+    };
+    this.previewForm = this.formBuilder.group(controls);
+  }
+
+  private getValidators(option: any): ValidatorFn[] {
+    const result = [];
+    switch (option.type) {
+      case 'int':
+        result.push(CdValidators.number());
+        result.push(Validators.required);
+        if (_.isNumber(option.min)) {
+          result.push(Validators.min(option.min));
+        }
+        if (_.isNumber(option.max)) {
+          result.push(Validators.max(option.max));
+        }
+        break;
+      case 'str':
+        if (_.isNumber(option.min)) {
+          result.push(Validators.minLength(option.min));
+        }
+        if (_.isNumber(option.max)) {
+          result.push(Validators.maxLength(option.max));
+        }
+        break;
+    }
+    return result;
+  }
+
+  private getReport() {
+    this.loading = true;
+    this.telemetryService.getReport().subscribe(
+      (resp: object) => {
+        this.report = resp;
+        this.reportId = resp['report']['report_id'];
+        this.createPreviewForm();
+        this.loading = false;
+        this.step++;
+      },
+      (_error) => {
+        this.error = true;
+      }
+    );
+  }
+
+  updateConfig() {
+    const config = {};
+    _.forEach(Object.values(this.options), (option) => {
+      const control = this.configForm.get(option.name);
+      // Append the option only if the value has been modified.
+      if (control.dirty && control.valid) {
+        config[option.name] = control.value;
+      }
+    });
+    this.mgrModuleService.updateConfig('telemetry', config).subscribe(
+      () => {
+        this.disableModule(
+          'Your settings have been applied successfully. ' +
+            'Due to privacy/legal reasons the Telemetry module is now disabled until you ' +
+            'completed the next step and accepted the license.',
```

Comment:
Wrong tense used here.
```suggestion
            'complete the next step and accept the license.',
```